packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -180,6 +205,26 @@ export const adjustInvalidDateSectionValue = <TDate, TSection extends FieldSecti
       return utils.formatByString(newDate, section.formatValue);
     }
 
+    case 'weekDay': {
+      let newDate: TDate;
+      if (shouldSetAbsolute) {
+        if (delta > 0 || isEnd) {
+          newDate = utils.startOfWeek(today);
```

Comment:
Indeed, it seems incoherent when pressing `Start`.

Do you agree that pressing `ArrowDown` on an empty day section should set `31` or should it set `1` ?
Spectrum sets `1`
Telerik sets `31`

In any case, I think we can fix it in another PR since the behavior is not linked only to the week days.

Reply:
Indeed, this fix can be done in a separate PR. 👍 

To me the following would seem most logical:
- <kbd>ArrowDown</kbd> -> `1`
- <kbd>ArrowUp</kbd> -> `31`

Reason behind it: "Up" sounds like it should increase the value, sort of makes sense that it would bump the value to max in case of empty section.

Reply:
The browser input sets `31` on arrow up.

But to take a kitchen example, when I press `-` on my electric hob, it sets the power to max whereas on my parent's one it sets to a low power.
Both have very good arguments.

Setting `1` is closer from the logic of the key pressed (arrow *down* so we sets to the value down in the list).
Setting `31` is more optimized to get to values close to the end of the month (to get to `29` I just have to do `ArrowDown` `ArrowDown` `ArrowDown` instead of `ArrowUp` `ArrowDown` `ArrowDown`, my find does not move. I personally find this behavior easier to use.

Reply:
> The browser input sets `31` on arrow up.

Well. Indeed, my assumption doesn't hold ground if the three major browsers do it the same we currently have as well. 🤔 
If major browsers did implement it this way, it's probably written in spec and we could safely go with the same implementation. 👌 
But at least we (I and browser native implementations) agree on the `Home` and `End` behaviour, where it's the same regardless of empty state or filled data.